app/liquid_tags/link_tag.rb

```diff
@@ -21,7 +21,15 @@ def get_article(slug)
   end
 
   def article_hash(slug)
-    path = Addressable::URI.parse(slug).path
+    url = Addressable::URI.parse(slug)
+    domain = url.port ? "#{url.host}:#{url.port}" : url.host
+    path = url.path
+
+    # If domain is present in url check if it belongs to the app
+    unless domain.blank? || domain&.casecmp?(SiteConfig.app_domain)
+      raise StandardError, "This URL is not an article link: {% link #{slug} %}"
```

Comment:
I have a feeling there is a better way to deal with this issue any thoughts?

Reply:
slug might not be a URL right? Then the message is slightly misleading

Reply:
when you use
```
{% link https://www.freecodecamp.org/news/ultimate-dynamodb-2020-cheatsheet/ %}
```
variable slug has value https://www.freecodecamp.org/news/ultimate-dynamodb-2020-cheatsheet/

and when you do
```
{% link abbottkarey/surprised-by-joy-et-assumenda-jgn %}
```
slug has the value **abbottkarey/surprised-by-joy-et-assumenda-jgn**

I think the variable name slug might be the cause of confusion. I was not sure what to rename it to so I just left it same.

Reply:
Probably I should rename slug to **url** or **url_string** 
(since there is ```    url = Addressable::URI.parse(slug)``` on the next line) ? any thoughts?

Reply:
I'm not talking about the variable though, I meant the wording of the error message ehhe

`/abbottkarey/surprised-by-joy-et-assumenda-jgn/` is not a URL so telling the user "the URL is not an article link" is misleading. Also because it's just a slug, not a link either.

Maybe something like "The article you're looking for does not exist" ?

Reply:
ok. I'll change it.